src/lib/transition_handler/processor.ml

```diff
@@ -43,6 +43,18 @@ let add_and_finalize ~logger ~frontier ~catchup_scheduler
   let transition =
     Transition_frontier.Breadcrumb.validated_transition breadcrumb
   in
+  (let metadata =
+     [ ( "state_hash"
+       , Transition_frontier.Breadcrumb.state_hash breadcrumb
+         |> State_hash.to_yojson )
+     ]
+   in
+   Option.value_map valid_cb
```

Comment:
NIT: the log function supports string interpolation, so it would probably be neater to just do `let existence = if Option.is_some valid_cb then "with" else "without"` and then interpolate that into the log statement `[%log debug] "add_and_finalize $state_hash %s callback" existence ~metadata`.